### PR TITLE
curl: add option to use wolfssl backend

### PR DESCRIPTION
**Pull request by malbarbo**
###### Motivation for this change
Allow smaller closure by using wolfssl backend.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @lovek323

### CODE REVIEW

**Comment by andir** [comment]
This should be targeted at the `staging` branch

File: pkgs/tools/networking/curl/default.nix [hunk 1]
```diff
@@ -67,10 +72,11 @@ stdenv.mkDerivation rec {
       # to nss-cacert from the default profile.
       "--without-ca-bundle"
       "--without-ca-path"
-      "--with-ca-fallback"
       "--disable-manual"
+      ( if sslSupport || gnutlsSupport then "--with-ca-fallback" else "--without-ca-fallback")
```
Comment (andir) [inline pkgs/tools/networking/curl/default.nix]: Doesn't this also apply for the wolfssl backend?

Comment (malbarbo) [reply pkgs/tools/networking/curl/default.nix]: The issue here is that wolfssl does not build with `--with-ca-fallback`, so we only pass `--with-ca-fallback` for openssl and gnutls.

**Comment by malbarbo** [comment]
Added a comment about ca fallback and try to avoid mass rebuild.

**Comment by malbarbo** [comment]
Rebuild is not needed anymore. @lovek323 can you take a look?

**Pull request merged**